ADR/0033-product-version-controller.md

```diff
@@ -0,0 +1,252 @@
+discussion of these terms.
+
+[oref]: https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/
+[pvsp]: https://opensource.com/article/19/11/product-vs-project
+
+### Streamlining version creation from the API/CLI.
+
+We will define the **ProjectDevStreamTemplate** custom resource. The *spec* for
+such a resource will include:
+
+* A **project** element referring to an existing *Project* resource.
+* A **variables** element that includes a list of variable names and optional
+  default values. The default values of variables my be defined in terms of
+  other, previously defined, variables using the Go template syntax.
+* A **resources** element that includes a list of resource definitions so that
+  each definition may reference variables by using the Go template syntax.
+
+Here is an example for what a *ProjectDevStreamTemplate* resource may look like:
+
+```
+apiVersion: pvc.konflux.dev/v1alpha1
+kind: ProjectDevStreamTemplate
+metadata:
+  name: template1
+  ...
+spec:
+  project: project1
+  variables:
+  - name: foo
+  - name: bar
+    defaultValue: "baz{foo}"
+  resources:
+  - apiVersion: appstudio.redhat.com/v1alpha1
+    kind: Application
+    metadata:
+      name: "{foo}-app1"
+    spec:
+      displayName: "{foo}-app1"
+  - apiVersion: appstudio.redhat.com/v1alpha1
+    kind: Component
+    metadata:
+      name: "{foo}-comp1"
+    spec:
+      application: "{foo}-app1"
+      componentName: "{foo}-comp1"
+      source:
+        git:
+          context: ./
+          dockerfileUrl: Dockerfile
+          revision: {bar}
+          uri: ...
+```
+
+The *spec* section for the *ProjectDevStream* may include a **template** section
+that includes a **name** field for referring to a *ProjectDevStreamTemplate*
+resource and a **values** section to provide values for the template variables.
+When this happens, the resources described by the template get created.
+
+Here is an example for what a *ProjectDevStream* resource with a template may
+look like:
+
+```
+apiVersion: pcv.konflux.dev/v1alpha1
+kind: ProjectDevStream
+metadata:
+  ...
+spec:
+  template:
+    name: template1
+    values:
+      - name: foo
+        value: some-value-here
+```
+
+The *values* section must provide values for all template variables without
+default values defined, it may also include values for variable that do have
+default values defined.
+
+The example above will cause the creation of the "some-value-here-app1"
+*Application* resource and the "some-value-here-comp1" *Component* resource.
+
+### Streamlining version cloning from the UI
+
+We will support a process for cloning a *ProjectDevStream* resource and all the
+resources included in it. While the process could be used from the API, it is
+primarily meant to be used from the Konflux UI.
+
+Several technical elements would be needed to support this process.
+
+First and foremost we will support a process for generating a
```

Comment:
I could use some CR examples like you have in the previous section. I am having a harder time following the flow here.

Reply:
Added a rather large example in the bottom.